coalib/bearlib/__init__.py

```diff
@@ -3,3 +3,45 @@
 as the rest of coala the bearlib is designed to be as easy to use as possible
 while offering the best possible flexibility.
 """
+
+from coalib.settings.FunctionMetadata import FunctionMetadata
+
+
+def deprecate_settings(**depr_args):
+    """
+     The purpose of this decorator is to allow passing old settings names to
+     bears due to the heavy changes in their names.
+
+     >>> @deprecate_settings(new='old')
+     ... def run(new):
+     ...     print(new)
+
+     Now we can simply call the bear with the deprecated setting, we'll get a
+     warning - but it still works!
+
+     >>> run(old="Hello world!")
+     The setting `old` is deprecated. Please use `new` instead.
+     Hello world!
+
+     :param depr_args:     A dictionary of settings as keys and their
+                           deprecated names as values.
+    """
+    def _deprecate_decorator(func):
+
+        def wrapping_function(*args, **kwargs):
+            for arg, deprecated_arg in depr_args.items():
+                if deprecated_arg in kwargs and arg not in kwargs:
```

Comment:
You should at least add a comment documenting that we can't do this linked deprecation thing